compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -33,28 +33,27 @@ pub fn find_native_static_library(
     search_paths: &[PathBuf],
     sess: &Session,
 ) -> PathBuf {
-    let verbatim = verbatim.unwrap_or(false);
-    // On Windows, static libraries sometimes show up as libfoo.a and other
-    // times show up as foo.lib
-    let oslibname = if verbatim {
-        name.to_string()
+    let formats = if verbatim.unwrap_or(false) {
+        vec![("".to_string(), "".to_string())]
     } else {
-        format!("{}{}{}", sess.target.staticlib_prefix, name, sess.target.staticlib_suffix)
+        let os =
+            (sess.target.staticlib_prefix.to_string(), sess.target.staticlib_suffix.to_string());
```

Comment:
```suggestion
            (sess.target.staticlib_prefix, sess.target.staticlib_suffix);
```
You can keep all the strings here as `Cow<'static, str>`s, no need to convert them to `String`s.